### PR TITLE
[IMP] account: Adding a method to check reconcilability on bank statements

### PR DESCRIPTION
**Pull request by hbto**
[IMP] account: Adding a method to check reconcilability on
account.bank.statement.line and make it inheritable.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by moylop260** [comment]
@nim-odoo 
Could you help us to check it, please?

File: addons/account/models/account_bank_statement.py [hunk 1]
```diff
@@ -539,8 +539,7 @@ def process_reconciliation(self, counterpart_aml_dicts=None, payment_aml_rec=Non
         counterpart_moves = self.env['account.move']
 
         # Check and prepare received data
-        if any(rec.statement_id for rec in payment_aml_rec):
-            raise UserError(_('A selected move line was already reconciled.'))
+        self.check_aml_reconcilability(payment_aml_rec)
```
Comment (nim-odoo) [inline addons/account/models/account_bank_statement.py]: It should be a private method.

**Comment by hbto** [comment]
@nim-odoo 

> 
What is the use case you are trying to solve?

Users want to add validations at the reconciliation of Journal Items of future or past dates than that of the Bank Statement Line date.

By adding this method it will be possible to add those validations without having to override the whole method.

Regards.

File: addons/account/models/account_bank_statement.py (hunk 1 above)
Comment (hbto) [reply addons/account/models/account_bank_statement.py]: Done on latest commit.

Regards

**Comment by nim-odoo** [comment]
I don't think this is in any way necessary. You can do:

```python
def process_reconciliation(self, counterpart_aml_dicts=None, payment_aml_rec=None, new_aml_dicts=None):
    for rec in payment_aml_rec or self.env['account.move.line']:
        [do whatever check you want here]
    super().process_reconciliation(counterpart_aml_dicts, payment_aml_rec, new_aml_dicts)
```

No valuable information is computed before the piece of code you want to override, only a couple of variable assignments. Making this inheritable won't make an override cleaner or easier.

**Comment by mart-e** [comment]
Closing following the reasons given by Nicolas.

**Pull request closed without merging**